src/components/Link.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+
+type LinkVariantProps = React.HTMLProps<HTMLAnchorElement> & {
+  variant?: "link";
+  href: string;
+};
+type NavVariantProps = React.HTMLProps<HTMLLIElement> & {
+  isActive: boolean;
+  variant: "nav";
+  onClick: () => void;
+};
+type Props = LinkVariantProps | NavVariantProps;
+
+/**
+ * A reusable link component with animated underline when the link is active.
+ *
+ * Has two variants:
+ *  1. "link" (default): renders an <a> tag with hover/focus styling.
+ *  2. "nav": renders a <button> with an onClick callback. Active state styling
+ *     is handled by an isActive prop, rather than directly by hover/focus.
+ */
+const Link = (props: Props) => {
+  const { variant, children } = props;
```

Comment:
I don't mind one bit as is, but this could be rewritten as follows,

```typescript
const Link = ({ children, variant, ...props }: Props) => {
```

Is it just a preference to destructure it after the function declaration? Or is there a function to it that I've missed? 😄

Reply:
Naw I don't mind one bit doing it one way or another. I just have a habit myself of destructuring the arguments directly. Personally I wouldn't even use the `...props` normally, I just didn't want to tamper with the rest of the code that uses `props`. 😆